File: Programmers/Lv.1/기사단원의 무기.py
def count(num):
    result = 0
    for i in range(1, int(num**0.5)+1):
        if i**2 == num:
            result += 1
        elif num % i == 0:
            result += 2
            
    return result

def solution(number, limit, power):
    answer = 0
    
    for i in range(1, number+1):
        div = count(i)
        print(div)
        if div > limit:
            answer += power
        else:
            answer += div
            
    return answer

File: Programmers/Lv.1/기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        
        int[] count = new int[number + 1];   
        
        // 일정 범위 내의 약수 구하기 테크닉
        for (int i = 1; i <= number; i++) {
            for (int j = 1; j <= number / i; j++) {
                count[i * j]++;
            }
        }
        
        int answer = 0;
        for (int i = 1; i <= number; i++) {
            if (count[i] > limit) {
                answer += power;
            } else {
                answer += count[i];
            }
        }
        
        return answer;
    }
}